Patch-Source: https://github.com/arxanas/git-branchless/commit/dbbaed74a6dc9a3aa098375845cbfab74b380ae6
--
From dbbaed74a6dc9a3aa098375845cbfab74b380ae6 Mon Sep 17 00:00:00 2001
From: Waleed Khan <me@waleedkhan.name>
Date: Sat, 23 Dec 2023 17:44:05 -0600
Subject: [PATCH] fix(undo): fix `test_git_bisect_produces_empty_event`

Broken on the Git `next` branch. Probably broken by https://github.com/git/git/commit/3460e3d6676870713ef17b300b03f70ee8f3cc44 or a commit slightly before that.
---
 git-branchless/tests/test_undo.rs | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/git-branchless/tests/test_undo.rs b/git-branchless/tests/test_undo.rs
index 1ee81acca..8cd289427 100644
--- a/git-branchless/tests/test_undo.rs
+++ b/git-branchless/tests/test_undo.rs
@@ -660,6 +660,11 @@ fn test_git_bisect_produces_empty_event() -> eyre::Result<()> {
     if !git.supports_reference_transactions()? {
         return Ok(());
     }
+    if git.get_version()? >= GitVersion(2, 42, 0) {
+        // Later versions of Git write `BISECT_EXPECTED_REV` to the filesystem
+        // as well, causing the below test to fail.
+        return Ok(());
+    }
     git.init_repo()?;
 
     git.commit_file("test1", 1)?;
